sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/utils/AutoValueUtils.java

```diff
@@ -139,11 +139,15 @@ private static boolean matchConstructor(
                 Collectors.toMap(
                     f -> ReflectUtils.stripGetterPrefix(f.getMethod().getName()),
                     Function.identity()));
-    for (FieldValueTypeInformation type : getterTypes) {
-      if (typeMap.get(type.getName()) == null) {
+
+    // Verify that constructor parameters match (name and type) the inferred schema.
+    for (Parameter parameter : constructor.getParameters()) {
+      FieldValueTypeInformation type = typeMap.getOrDefault(parameter.getName(), null);
+      if (type == null || type.getRawType() != parameter.getType()) {
```

Comment:
I'm not quite sure what the old logic was trying to verify. Now this verifies that every constructor parameter corresponds to a schema field.

Reply:
I believe the getterTypes is the list of schema name, where typeMap is the attribute name from AutoValue class. It's trying to verify whether the schema matches the AutoValue class since it wants to get a constructor.